athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDbIntegTest.java

```diff
@@ -47,20 +53,87 @@ public class DynamoDbIntegTest extends IntegrationTestBase {
+        tableUtils = new DdbTableUtils();
+    }
+
+    /**
+     * Insert rows into the newly created DDB table.
+     */
+    protected void addMovieItems()
+    {
+        tableUtils.putItem(movieTableName, createMovieItem("2014", "Interstellar", "Christopher Nolan",
+                ImmutableList.of("Matthew McConaughey", "John Lithgow", "Ann Hathaway",
+                        "David Gyasi", "Michael Caine", "Jessica Chastain",
+                        "Matt Damon", "Casey Affleck"),
+                ImmutableList.of("Adventure", "Drama", "Sci-Fi", "Thriller")));
+        tableUtils.putItem(movieTableName, createMovieItem("1986", "Aliens", "James Cameron",
+                ImmutableList.of("Sigourney Weaver", "Paul Reiser", "Lance Henriksen",
+                        "Bill Paxton"),
+                ImmutableList.of("Adventure", "Action", "Sci-Fi", "Thriller")));
+    }
+
+    private void addDatatypeItems()
+    {
+        Map<String, AttributeValue> item = new ImmutableMap.Builder<String, AttributeValue>()
+            .put("int_type", new AttributeValue().withN(String.valueOf(TEST_DATATYPES_INT_VALUE)))
+            .put("smallint_type", new AttributeValue().withN(String.valueOf(TEST_DATATYPES_SHORT_VALUE)))
+            .put("bigint_type", new AttributeValue().withN(String.valueOf(TEST_DATATYPES_LONG_VALUE)))
+            .put("varchar_type", new AttributeValue(TEST_DATATYPES_VARCHAR_VALUE))
+            .put("boolean_type", new AttributeValue().withBOOL(TEST_DATATYPES_BOOLEAN_VALUE))
+            .put("float4_type", new AttributeValue(String.valueOf(TEST_DATATYPES_SINGLE_PRECISION_VALUE)))
+            .put("float8_type", new AttributeValue(String.valueOf(1E-130))) // smallest number dynamo can handle
+            .put("date_type", new AttributeValue(TEST_DATATYPES_DATE_VALUE))
+            .put("timestamp_type", new AttributeValue(TEST_DATATYPES_TIMESTAMP_VALUE))
+            .put("byte_type", new AttributeValue().withB(ByteBuffer.wrap(TEST_DATATYPES_BYTE_ARRAY_VALUE)))
+            .put("textarray_type", new AttributeValue(TEST_DATATYPES_VARCHAR_ARRAY_VALUE)).build();
+        tableUtils.putItem(TEST_DATATYPES_TABLE_NAME, item);
     }
 
 
+    /**
+     * Creates and returns a Map containing the item's attributes and adds the item to the table.
+     * @param year Year attribute.
+     * @param title Title attribute.
+     * @param director Director attribute.
+     * @param cast List of cast members.
+     * @param genre List of movie generes.
+     */
+    private Map<String, AttributeValue> createMovieItem(String year, String title, String director,
+            List<String> cast, List<String> genre)
+    {
+        logger.info("Add item: year=[{}], title=[{}], director=[{}], cast={}, genre={}",
+                year, title, director, cast, genre);
+
+        List<AttributeValue> castAttrib = new ArrayList<>();
+        cast.forEach(value -> castAttrib.add(new AttributeValue(value)));
```

Comment:
Haven't used Java in a while, but could this be done as follows instead? (coming from scala so not sure on the exact java syntax)

List<AttributeValue> castAttrib = cast.stream().map(value -> new AttributeValue(value)).collect(Collectors.toList());

Reply:
This code is moved not changed, so I didn't really touch it. In general yeah we could probably make this more "functional/streaming java" but since this isn't technically a change but a refactor here, I will defer that to a future PR.